### PR TITLE
Appointment scheduling

### PR DESCRIPTION
**Pull request by qangdev**
- Added max slot per service in cmd/main.py
- Load booked date in appointment route to handle time slot on the front end
- Some other small changes

### CODE REVIEW

File: app/models/services.py [hunk 1]
```diff
@@ -26,6 +26,7 @@ def insert(shop_id="", name="", description="", prices="", images=None):
             "description": description,
             "prices": prices,
             "images": images or [],
+            "max_slot": max_slot
```
Comment (mohanadkaleia) [inline app/models/services.py:29]: nicr

File: app/views/home.py [hunk 2]
```diff
@@ -61,7 +64,7 @@ def send_email():
     from_email = request.form["from_email"]
     subject = request.form["subject"]
     content = request.form["content"]
-    config = get_config()
+    # config = get_config()  # This one seems already defined at the top
```
Comment (mohanadkaleia) [inline app/views/home.py]: good catch! lets remove it then

File: app/views/home.py [hunk 3]
```diff
@@ -72,13 +75,55 @@ def send_email():
...
+        email=request.form.get("email"),
+        description=request.form.get("description")
+    )
+    sendgrid.send(
+        config["TO_EMAIL_ADDRESS"],
+        request.form.get("email"),
+        "Your appointment has been received",
+        "Your appointment has been received"
+    )
+    sendgrid.send(
```
Comment (mohanadkaleia) [inline app/views/home.py]: I think no need to send the customer an email here, can we remove this one

Comment (qangdev) [reply app/views/home.py]: ok cool! I've pushed an update for these changes :)

File: app/views/home.py [hunk 4]
```diff
@@ -72,13 +74,49 @@ def send_email():
...
+        "services": services,
+        "appointments": timeslots,
+        "locked_slots": locked_slots
+    }
     return render_template("appointment.html", config=config, data=data)
 
 
+@bp.route("/appointment", methods=["POST"])
+def save_appointment():
+    appointment_model.insert(
```
Comment (mohanadkaleia) [inline app/views/home.py]: found one issue here, if the insertion failed an email would still be sent. I think we should do some error handling here

File: app/views/home.py [hunk 5]
```diff
@@ -72,13 +74,49 @@ def send_email():
...
+        vehicle=request.form.get("vehicle"),
+        name=request.form.get("name"),
+        email=request.form.get("email"),
+        description=request.form.get("description")
+    )
+    sendgrid.send(
+        config["TO_EMAIL_ADDRESS"],
+        config["TO_EMAIL_ADDRESS"],
+        "New appointment has been created",
+        "New appointment has been created"
```
Comment (mohanadkaleia) [inline app/views/home.py]: I think the body should be something different than just a "new appointment has been created" we should add more details like the service name, and all other information

**Pull request merged**